src/coreclr/src/vm/methodtable.cpp

```diff
@@ -9949,6 +9929,54 @@ NOINLINE BYTE *MethodTable::GetLoaderAllocatorObjectForGC()
     return retVal;
 }
 
+UINT32 MethodTable::GetNativeSize()
+{
+    CONTRACTL
+    {
+        THROWS;
+        MODE_ANY;
+        PRECONDITION(CheckPointer(GetClass()));
+    }
+    CONTRACTL_END;
+    if (IsBlittable())
+    {
+        return GetClass()->GetLayoutInfo()->GetManagedSize();
+    }
+    return GetNativeLayoutInfo()->GetSize();
+}
+
+EEClassNativeLayoutInfo const* MethodTable::GetNativeLayoutInfo()
+{
+    CONTRACTL
+    {
+        THROWS;
+        MODE_ANY;
+        PRECONDITION(HasLayout());
+    }
+    CONTRACTL_END;
+    EEClassNativeLayoutInfo* pNativeLayoutInfo = GetClass()->GetNativeLayoutInfo();
+    if (pNativeLayoutInfo != nullptr)
+    {
+        return pNativeLayoutInfo;
+    }
+    EnsureNativeLayoutInfoInitialized();
+    return GetClass()->GetNativeLayoutInfo();
```

Comment:
Nit: It is nice to put the re-fetch of the value after it is initialized into the slow-path function. It helps to get it inlined the way you would like.